src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -211,11 +200,22 @@ private unsafe int CompareString(ReadOnlySpan<char> string1, string string2, Com
 
             string? localeName = _sortHandle != IntPtr.Zero ? null : _sortName;
 
+            // CompareStringEx may try to dereference the first character of its input, even if an explicit
+            // length of 0 is specified. To work around potential AVs we'll always ensure zero-length inputs
+            // are normalized to a null-terminated empty string.
```

Comment:
Do we have test coverage for all these empty / null string corner cases?

Reply:
Yes. The unit tests that are part of https://github.com/dotnet/runtime/pull/1514 (from which this PR was broken out as a more manageable chunk) cover this scenario. In fact, those unit tests are where the problem was originally discovered. :)